video/out/drm_common.c

```diff
@@ -588,6 +628,84 @@ static bool setup_crtc(struct vo_drm_state *drm, const drmModeRes *res)
     return true;
 }
 
+static void setup_edid(struct vo_drm_state *drm)
+{
+    drmModePropertyBlobRes *blob;
+    for (int i = 0; i < drm->connector->count_props; ++i) {
+        drmModePropertyRes *prop = drmModeGetProperty(drm->fd, drm->connector->props[i]);
+        if (prop && strcmp(prop->name, "EDID") == 0) {
+            blob = drmModeGetPropertyBlob(drm->fd, drm->connector->prop_values[i]);
+        }
+        drmModeFreeProperty(prop);
+        if (blob)
+            break;
+    }
+    if (!blob) {
+        MP_VERBOSE(drm, "Unable to get EDID blob from connector.\n");
+        return;
+    }
+
+    drm->info = di_info_parse_edid(blob->data, blob->length);
+    if (!drm->info) {
+        MP_VERBOSE(drm, "Failed to parse EDID info: %s\n", mp_strerror(errno));
+        goto done;
+    }
+
+    const struct di_edid *edid = di_info_get_edid(drm->info);
+    if (!edid) {
+        MP_VERBOSE(drm, "Failed to get EDID info: %s\n", mp_strerror(errno));
+        goto done;
+    }
+
+    drm->chromaticity = di_edid_get_chromaticity_coords(edid);
+
+    const struct di_edid_cta *cta;
+    const struct di_edid_ext *const *exts = di_edid_get_extensions(edid);
+    for (int i = 0; exts[i]; ++i) {
+        if ((cta = di_edid_ext_get_cta(exts[i])))
+            break;
+    }
```

Comment:
It is fine, but probably something like would work too.
```suggestion
    while (*exts && !(cta = di_edid_ext_get_cta(*exts++)));
```

Reply:
That should work.